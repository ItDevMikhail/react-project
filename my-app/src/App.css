.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
h2{
  text-align: center;
}
.registerForm,
.loginForm{
  display: flex;
  flex-direction: column;
  max-width: 300px;
  align-items: center;
  margin: 0 auto;
}
.registerForm input,
.loginForm input{
  width: 280px;
}
.regErrors{
  color: red;
}
.inputErrors{
  border-bottom: 1px solid red;
}
.registerCard,
.loginCard{
  max-width: 320px;
  margin: 100px auto;
}
svg.MuiSvgIcon-root {
 width: auto;
 height: 36px;
 border-radius: 5px;
 fill: #000;
 background-color:#e0e0e0;
}
svg.MuiSvgIcon-root:hover{
  fill: #e0e0e0;
  background-color:#000;
}
#iconButton.MuiButton-root{
  min-width: 0;
  line-height: 0;
}
.registerCardHeader,
.loginCardHeader{
  text-align: center;
}
.header{
 display: flex;
 justify-content: space-between;
 align-items: center;
 background-color: #3f51b5;
 height: 100px;
}
.header a{
  text-decoration: none;
  color: #000;
}
.navMenuRight svg{
  height: 36px;
  width: 50px;
  fill: #000;
  border-radius: 5px;
  background-color: #e0e0e0;
}
.navMenuRight svg:hover{
  fill: #e0e0e0;
  background-color: #000;
}
#iconButton{
  padding: 0;
  background-color: transparent;
}
.selected #iconButton svg{
  fill: #e0e0e0;
  background-color: #000;
}
.navMenuLeft .selected button{
  background-color: #000;
  color: #fff;
}
.navMenuLeft a + a{
  margin-left: 20px;
}