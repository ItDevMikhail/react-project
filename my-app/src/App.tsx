import React, { useState, useEffect } from 'react';
import './App.css';
import { BrowserRouter as Router, Route, NavLink, Redirect, Switch } from 'react-router-dom';
import LoginPage from './pages/LoginPage';
import RegisterPage from './pages/RegisterPage';
import LibraryPage from './pages/LibraryPage';
import RootGaurd from './commponents/guard/auth.guard';
import AddBookPage from './pages/addBookPage/index';
import HomePage from './pages/homePage';
import BookDetailPage from './pages/bookDetailPage';
import { useHttp } from './hooks/http.hook';
import { useHttpAuth } from './hooks/http.auth.hook';




function App() {
  const { request, error } = useHttp();
  const [isLogged, setIsLogged] = useState(true);
  const { requestAuth, errorMess } = useHttpAuth();
  const logged = (val: boolean) => {
    setIsLogged(val);
  }
  const guard = () => {
    return isLogged;
  }
  const logout = async () => {
    setIsLogged(false);
    try {
      const data = await request('/api/users/logout', 'GET');
      if (data) {
        sessionStorage.clear();
      }
    } catch (e) {
      console.log(error);
    }
  }
  const checkauth = async () => {
    try {
      const data = await requestAuth();
      if (data) {
        setIsLogged(data.access);
      } else {
        setIsLogged(false);
      }
    } catch (error) {
      setIsLogged(false);
      console.log(errorMess);
    }
  }
  useEffect(() => {
    checkauth();
  }, []);

  return (
    <Router>
      <RootGaurd logged={guard}>
        <div className="header">
          <div className="container">
            {isLogged && <nav className="navMenu"><div className="navMenuleft">
              <NavLink activeClassName="selected" to='/home' exact={true}><span className="material-icons">
                home
              </span></NavLink>
              <NavLink activeClassName="selected" to='/library' exact={true}>Library</NavLink>
              <NavLink activeClassName="selected" to='/library/add' exact={true}>Add book</NavLink>
            </div><div className="navMenuRight"><NavLink activeClassName="selected" to='/login' onClick={logout}><span className="material-icons">
              logout
            </span></NavLink></div></nav>}
            {!isLogged && <nav className="navMenu"><div></div><div className="navMenuRight"><NavLink activeClassName="selected" to='/login'><span className="material-icons">
              login
            </span></NavLink>
              <NavLink activeClassName="selected" to='/register'><span className="material-icons">
                person_add_alt
              </span></NavLink></div></nav>}
          </div>
        </div>
        <Switch>
          <Route path='/home' >
            <HomePage />
          </Route>
          <Route path='/library/detail/:id' >
            <BookDetailPage />
          </Route>
          <Route path='/login' >
            <LoginPage logged={logged} />
          </Route>
          <Route path='/register'>
            <RegisterPage />
          </Route>
          <Route path='/library' exact={true}>
            <LibraryPage />
          </Route>
          <Route path='/library/add'>
            <AddBookPage />
          </Route>
          <Redirect to='/login' />
        </Switch>
      </RootGaurd>
    </Router>
  );
}

export default App;
