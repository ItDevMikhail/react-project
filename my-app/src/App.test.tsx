import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders correctly', () => {
  const app = render(<App />);
  expect(app).toMatchSnapshot();
  // ругается на импорт fancybox т.к. в TS он задекларирован отдельным модулем. Для проверки теста нужно закомментировать fancybox в bookdetailpage

});
