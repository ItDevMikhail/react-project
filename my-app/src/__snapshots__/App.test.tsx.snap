// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="container"
        >
          <nav
            class="navMenu"
          >
            <div
              class="navMenuleft"
            >
              <a
                aria-current="page"
                class="selected"
                href="/home"
                title="Home"
              >
                <span
                  class="material-icons"
                >
                  home
                </span>
              </a>
              <a
                href="/library"
              >
                Library
              </a>
              <a
                href="/library/add"
              >
                Add book
              </a>
              <a
                href="/blog"
              >
                Blog
              </a>
              <a
                href="/user/dashboard"
              >
                Dashboard
              </a>
            </div>
            <div
              class="navMenuRight"
            >
              <a
                class=""
                href="/login"
                title="logout"
              >
                <span
                  class="material-icons"
                >
                  logout
                </span>
              </a>
            </div>
          </nav>
        </div>
        
      </div>
      <div
        class="container"
        style="background-color: rgb(235, 235, 235);"
      >
        <h2>
          Home page
        </h2>
        <h3>
          Здесь пока ничего нет кроме "Home"
        </h3>
        <img
          alt="images"
          class="homeIMG"
          src="home.jpg"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="container"
      >
        <nav
          class="navMenu"
        >
          <div
            class="navMenuleft"
          >
            <a
              aria-current="page"
              class="selected"
              href="/home"
              title="Home"
            >
              <span
                class="material-icons"
              >
                home
              </span>
            </a>
            <a
              href="/library"
            >
              Library
            </a>
            <a
              href="/library/add"
            >
              Add book
            </a>
            <a
              href="/blog"
            >
              Blog
            </a>
            <a
              href="/user/dashboard"
            >
              Dashboard
            </a>
          </div>
          <div
            class="navMenuRight"
          >
            <a
              class=""
              href="/login"
              title="logout"
            >
              <span
                class="material-icons"
              >
                logout
              </span>
            </a>
          </div>
        </nav>
      </div>
      
    </div>
    <div
      class="container"
      style="background-color: rgb(235, 235, 235);"
    >
      <h2>
        Home page
      </h2>
      <h3>
        Здесь пока ничего нет кроме "Home"
      </h3>
      <img
        alt="images"
        class="homeIMG"
        src="home.jpg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
